### ISSUE TITLE
Auth - add anonymous provider as login

### ISSUE THREAD

**Issue by Scrat94** [firebase/FirebaseUI-Android#900]
* Android device: Oneplus 3T 
  * Android OS version: 7.1.1
  * Google Play Services version: 11.5.09
  * Firebase/Play Services SDK version: 3.1.0
  * FirebaseUI version: 2.3.0
  
### Feature Request:
Could you add Anonymous Provider for login. I would like to have the possibility that my user can login anonymously.

I would appreciate something like this:
```
[...]
new AuthUI.IdpConfig.Builder(AuthUI.ANONYMOUS_PROVIDER).build())
[...]
```

**Comment by samtstern** [comment]
@Scrat94 thanks for the request!  We don't plan to support this in FirebaseUI because anonymous login should not need a UI at all.  You can anonymously log in your users as soon as they open the app, and then present them the option to upgrade to a real account at a later time.

**Comment by SUPERCILEX** [comment]
@Scrat94 also, see #123.

**Comment by samtstern** [comment]
@Scrat94 we changed our minds!  See #1375

**Comment by samtstern** [comment]
This was fixed and released in version `4.2.0`

**Comment by iamdeadman** [comment]
![screenshot 2018-12-23 at 1 27 38 am](https://user-images.githubusercontent.com/3391877/50378192-1d4a3c00-0652-11e9-91a1-fcc7b98100ea.png)
But the current released version is 4.1.0 according to docs ? and 4.2.0 does not seem to be there in maven atleast I think. Do you have information regarding 4.2.0 release date ?
